Denote paid/premium features in the manual

pjhul opened the pull request:
This PR denotes which features are available on which plans. Previously, this information was either inconsistent or unavailable, and gathering this information took _much_ longer than I expected, but having it accessible will be a huge plus.

Deploy preview status seems not to be updating, so here's the [link](https://deploy-preview-4158--posthog.netlify.app/).

## Changes
- Displays feature availability in sidebar component
<img width="263" alt="Screen Shot 2022-09-13 at 2 17 47 AM" src="https://user-images.githubusercontent.com/39424187/189866605-bd2a6da0-4148-4e09-b3b8-b05c025de968.png">
- Displays the availability of sub-features within each manual page
<img width="746" alt="Screen Shot 2022-09-13 at 2 34 37 AM" src="https://user-images.githubusercontent.com/39424187/189867022-e2a05ccb-8a93-4887-a8cc-45a81fc8ee11.png">

## To-do
- [x] Multiple organizations and Google SSO are available on free cloud but not on free self-host, need to denote this

## Inconsistencies
- [x] ~~Cohorts are listed as :warning: but it's not clear which features are restricted~~ Cohort based on 'completed sequence of events' and 'Lifecycle' criteria
- [x] ~Data management is listed as :warning: but it's not clear which features are restricted~ Tagging and adding descriptions
- [x] 'Event & properties taxonomy' is listed as not available on FOSS on the pricing page, but it seems as though it is
- [x] What are 'Advanced user permissions' listed on the Pricing page
- [x] Double checking SSO as I did not have time to verify all those
- [x] Google SSO is listed as not available on open-source, but this seems wrong
- [x] User permissions is listed as unavailable on FOSS, but it seems that it is

Closes #4062

pjhul commented:
@corywatilo This should at least be ready for you to take a look at the overall UX. The last major thing I have left to do is to denote that multiple organizations is available on the free cloud plan but _not_ on the OS version. I missed this early since it seems like this is the _only_ feature that is like this (frustratingly), so I'll need figure out how best to denote this. I'm also planning on getting a couple of product people to give this a once-over just to check that all the features are correct 👍

corywatilo reviewed (changes requested):
Whoa, a ton of work here! 🙌🏼 Thanks for slogging through all of this - vastly needed.

There's only one weird UX things to me here, but I think there are some easy fixes.

## Problem

In the right rail, where not all features are available, it shows the ⚠️ icon. But a quick scan of this page shows ❌ everywhere it breaks out feature availability inline.

<img width="1364" alt="image" src="https://user-images.githubusercontent.com/154479/192121785-bbf0543f-8aac-4868-8be9-a1df861860b7.png">

Because I'm smart (or at least that's what my mom told me), I'm _deducing_ that _all_ features are available, up until the point they're broken down. Assuming this is accurate, I propose:

1. Add a tooltip to the ⚠️ icon that says something like, "Not all features are available on this plan. See the availability table for details where features aren't available." (It's wordy, but you get the idea...)
2. Split the page into two sections: _Available on all plans_ (which shows check marks next to all plans) | _Available on some plans_ (where each feature has the breakdown) -- and a huge divider between the two

Alternatively/in addition, it might be nice to encompass each inline availability with a dashed line so we can see where the section ends.

I think this should make it clear enough about where certain plans don't get everything. (For features where _no_ features are available on all plans (eg [Group analytics](https://deploy-preview-4158--posthog.netlify.app/manual/group-analytics)), the inline breakdown should be at the top of the page.

<img width="1240" alt="image" src="https://user-images.githubusercontent.com/154479/192121951-b2bbee05-a62a-4358-a7f5-dd16fa89ad5f.png">

pjhul commented:
Alright, so I went through and made two changes that I think should address these issues:
1. I've added a tooltip to the sidebar feature availability that explains more about what each icon means (e.g. "Not all parts of this feature are available on this plan").
2. On the dashboards page, I've gone through and added a couple feature availability sections to make it more obvious that basic things like creating and editing dashboards are completely available on all plans. 

I've decided not to go with the idea of breaking each manual page up into separate sections, mainly because this would require a time commitment that I don't think is worth it for me right now, especially given how long this change has been in limbo and how many things I have in the pipeline coming up. This is already a very large change and so I think it's worth just going ahead and merging since it will already make a huge difference in its current state, and we can fine-tune things later. 

@corywatilo I think I'll merge this unless you have any major objections?

pjhul commented:
@corywatilo Totally agree, just made those changes! We can take a last look during our 1:1 and should be ready to ship.
